collect_app/src/main/java/org/odk/collect/android/fragments/dialogs/FormsDownloadResultDialog.kt

```diff
@@ -0,0 +1,74 @@
+package org.odk.collect.android.fragments.dialogs
+
+import android.app.Dialog
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import org.jetbrains.annotations.TestOnly
+import org.odk.collect.android.R
+import org.odk.collect.android.formmanagement.downloaderror.FormsDownloadErrorActivity
+import org.odk.collect.android.formmanagement.downloaderror.FormsDownloadErrorItem
+import java.util.ArrayList
+
+class FormsDownloadResultDialog : DialogFragment() {
+    companion object {
+        const val FAILURES = "FAILURES"
+        const val NUMBER_OF_ALL_FORMS = "NUMBER_OF_ALL_FORMS"
+    }
+
+    private lateinit var failures: ArrayList<FormsDownloadErrorItem>
+    private var numberOfAllForms = 0
+
+    private lateinit var listener: FormDownloadResultDialogListener
+
+    interface FormDownloadResultDialogListener {
+        fun onCloseDownloadingResult()
+    }
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        if (context is FormDownloadResultDialogListener) {
+            listener = context
+        }
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        isCancelable = false
```

Comment:
I'm not sure if we need to block the user from dismissing this with back/clicking outside. What do you think?

Reply:
Fixed.